### PR TITLE
Add -1 as max_tokens option to OpenAI llm to request as many tokens as possible while keeping context below 4000

### PR DESCRIPTION
**Pull request by thepok**

### CODE REVIEW

File: langchain/llms/openai.py [hunk 1]
```diff
@@ -116,9 +116,40 @@ def __call__(self, prompt: str, stop: Optional[List[str]] = None) -> str:
...
+        # import the GPT-3 tokenizer
+        from transformers import GPT2TokenizerFast
+
+        # create a GPT-3 tokenizer instance
+        tokenizer = GPT2TokenizerFast.from_pretrained('gpt2')
+        
+        # tokenize the text using the GPT-3 tokenizer
+        tokenized_text = tokenizer.tokenize(prompt)
+
+        # calculate the number of tokens in the tokenized text (some OpenAI models have a max of 8,000 tokens?!?)
```
Comment (andersenchen) [inline langchain/llms/openai.py:153]: The max sequence length does vary per model! Might be useful to use a mapping here (and yes, Codex goes up to 8k!)

File: langchain/llms/openai.py [hunk 2]
```diff
@@ -116,9 +116,40 @@ def __call__(self, prompt: str, stop: Optional[List[str]] = None) -> str:
...
+        Returns:
+            The maximum number of tokens to generate for a prompt.
+
+        Example:
+            .. code-block:: python
+
+                max_tokens = openai.max_token_for_prompt("Tell me a joke.")
+        """
+        # import the GPT-3 tokenizer
+        from transformers import GPT2TokenizerFast
```
Comment (hwchase17) [inline langchain/llms/openai.py:145]: nit: can we do a try/except around this import statement (just to throw a more user friendly error) like in https://github.com/hwchase17/langchain/blob/5267ebce2d9144ef0b530ae04434008fc5618d8d/langchain/text_splitter.py#L64

File: langchain/llms/openai.py (hunk 1 above)
Comment (hwchase17) [reply langchain/llms/openai.py:153]: +1 to this! lets do some mapping for the common models, but then we can have a default so if a models not in there or a new one comes out we dont break

**Comment by thepok** [comment]
https://beta.openai.com/docs/models

**Comment by Uzay-G** [comment]
This doesn't seem to work for OpenAIChat models, would it be hard to add?

**Pull request merged**